### DISCUSSION TITLE
How to combine propqueries?

### DISCUSSION THREAD

**Discussion by selvanair** [openssl/openssl#23676] [Q&A] [answered]
I have  built-in provider in an application that should be used only for signing when the private key is external. So I set a propquery of "?provider!=myprovider" to discourage fetching from it. I also load user specified providers or the default provider. These could be loaded via openssl.cnf or explicitly and may require their own propquery.

Is there an API for merging propqueries? Also if the user-defined propquery comes from openssl.cnf, how do I read it from the libctx? Is there something like `EVP_get_default_properties()`?

**Comment by mattcaswell** [comment, marked as answer]
Unfortunately there is no public API available for merging property queries. There is also no API for querying the default properties. These would be possible to add (much of the support for it exists within libcrypto - its just not exposed) - but they're not there at the moment.
